agrona/src/main/java/org/agrona/concurrent/status/AtomicCounter.java

```diff
@@ -220,7 +220,26 @@ public long incrementOrdered()
         final long offset = addressOffset;
         final long currentValue = UnsafeApi.getLong(array, offset);
         UnsafeApi.putLongRelease(array, offset, currentValue + 1);
+        return currentValue;
+    }
 
+    /**
+     * Increment the counter.
+     * <p>
+     * This method is not atomic and this can lead to lost-updates due to race conditions. This load and store
+     * have plain memory semantics.
+     * <p>
+     * The typical use-case for this method is when writer and reader are the same thread.
+     *
+     * @return the previous value of the counter
+     * @since 2.1.0
+     */
+    public long incrementPlain()
+    {
+        final byte[] array = byteArray;
+        final long offset = addressOffset;
+        final long currentValue = UnsafeApi.getLong(array, offset);
+        UnsafeApi.putLong(array, offset, currentValue + 1);
         return currentValue;
     }
```

Comment:
Needs a CAS loop similar to `jdk.internal.misc.Unsafe#getAndAddLongRelease`:
```java
    @ForceInline
    public final long getAndAddLongRelease(Object o, long offset, long delta) {
        long v;
        do {
            v = getLong(o, offset);
        } while (!weakCompareAndSetLongRelease(o, offset, v, v + delta));
        return v;
    }
```

Reply:
A cas loop only makes sense when there are concurrent updates. Plain can only be used be used single-threaded.